Source/WebCore/rendering/EventRegion.cpp

```diff
@@ -209,8 +209,42 @@ bool EventRegionContext::shouldConsolidateInteractionRegion(IntRect bounds, Rend
     return false;
 }
 
+void EventRegionContext::addOcclusionMarginsForSmallInteractionRegions()
+{
+    constexpr int minimumSize = 20;
+    constexpr int occlusionMargin = 10;
+    Vector<InteractionRegion> interactionRegionsWithOcclusions;
+    for (auto& interactionRegion : m_interactionRegions) {
+        bool isSmallRect = false;
+        auto occlusionRect = interactionRegion.rectInLayerCoordinates;
+
+        if (interactionRegion.rectInLayerCoordinates.width() < minimumSize) {
+            occlusionRect.inflateX(occlusionMargin);
+            isSmallRect = true;
+        }
+        
+        if (interactionRegion.rectInLayerCoordinates.height() < minimumSize) {
+            occlusionRect.inflateY(occlusionMargin);
+            isSmallRect = true;
+        }
```

Comment:
since the conditions here are based on static values and the rect coordinates, wouldn't it be easier to do in as-we-go? [here](https://github.com/WebKit/WebKit/blob/a8fc73077082ccee3003ba4eaab65cfb23d9c37b/Source/WebCore/rendering/EventRegion.cpp#L151)

* we're already in an `Type::Interaction` branch
* we can append the shield first to get the expected order
* it'll save us one copy of the whole Vector

For the case where we have multiple rects we just need to make sure to append all shields (Occlusions) before we append the Interactions.